packages/rulesets/src/spectral/test/lro-delete-provisioning-state-specified.test.ts

```diff
@@ -0,0 +1,442 @@
+            type: "integer",
+            format: "int32",
+            minimum: 0,
+            maximum: 255,
+          },
+          id: {
+            type: "string",
+          },
+        },
+      },
+      FooResource: {
+        "x-ms-azure-resource": true,
+        properties: {
+          provisioningState: {
+            type: "string",
+            enum: ["Creating", "Canceled", "Deleting", "Failed"],
+          },
+        },
+      },
+      FooRule: {
+        type: "object",
+        properties: {
+          properties: {
+            $ref: "#/definitions/FooResource",
+            "x-ms-client-flatten": true,
+          },
+        },
+        required: ["properties"],
+      },
+    },
+  }
+  return linter.run(oasDoc).then((results) => {
+    expect(results.length).toBe(0)
+  })
+})
+
+test("ProvisioningStateSpecifiedForRegularDelete should find no errors", () => {
+  const oasDoc = {
+    swagger: "2.0",
+    paths: {
+      "/foo": {
+        delete: {
+          operationId: "Foo_Update",
+          description: "Test Description",
+          parameters: [
+            {
+              name: "foo_delete",
+              in: "body",
+              schema: {
+                $ref: "#/definitions/FooRequestParams",
+              },
+            },
+          ],
+          responses: {
+            "200": {
+              description: "Success",
+              schema: {
+                $ref: "#/definitions/FooRule",
+              },
+            },
+          },
+        },
+      },
+    },
+    definitions: {
+      FooRequestParams: {
+        allOf: [
+          {
+            $ref: "#/definitions/FooProps",
+          },
+        ],
+      },
+      FooProps: {
+        properties: {
+          servicePrecedence: {
+            description:
+              "A precedence value that is used to decide between services when identifying the QoS values to use for a particular SIM. A lower value means a higher priority. This value should be unique among all services configured in the mobile network.",
+            type: "integer",
+            format: "int32",
+            minimum: 0,
+            maximum: 255,
+          },
+          id: {
+            type: "string",
+          },
+        },
+      },
+      FooResource: {
+        "x-ms-azure-resource": true,
+        properties: {
+          provisioningState: {
+            type: "string",
+            enum: ["Creating", "Canceled", "Deleting", "Failed"],
+          },
+        },
+      },
+      FooRule: {
+        type: "object",
+        properties: {
+          properties: {
+            $ref: "#/definitions/FooResource",
+            "x-ms-client-flatten": true,
+          },
+        },
+        required: ["properties"],
+      },
+    },
+  }
+  return linter.run(oasDoc).then((results) => {
+    expect(results.length).toBe(0)
+  })
+})
+
+test("ProvisioningStateSpecifiedForLRODelete without schema should find no errors", () => {
```

Comment:
>should find no errors [](http://example.com/codeflow?start=60&length=21)

This should have violated some other rule that states that both 200 and 204 are required for sync deletes. Can u doublecheck that thats happening?

Reply:
I dont think if other rules would be considered in individual tests

Reply:
Also, I see 
 Per ARM Specs, all DELETE methods (non-async) must have responses code implementation: 200, 204.(RPC-Delete-V1-01) 
 has been implemented